src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -0,0 +1,374 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {Platform} from '@angular/cdk/platform';
+import {
+  AfterViewInit,
+  Attribute,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  Optional,
+  Output,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {
+  MAT_CHECKBOX_CLICK_ACTION,
+  MatCheckboxClickAction,
+  RippleRenderer,
+  RippleTarget,
+  ThemePalette
+} from '@angular/material';
+import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+import {MDCCheckboxAdapter, MDCCheckboxFoundation} from '@material/checkbox';
+
+let nextUniqueId = 0;
+
+const rippleTarget: RippleTarget = {
+  rippleConfig: {
+    radius: 20,
+    centered: true,
+    animation: {
+      // TODO(mmalerba): Use the MDC constants once they are exported separately from the
+      // foundation. Grabbing them off the foundation prevents the foundation class from being
+      // tree-shaken.
+      enterDuration: 225 /* MDCRippleFoundation.numbers.DEACTIVATION_TIMEOUT_MS */,
+      exitDuration: 150 /* MDCRippleFoundation.numbers.FG_DEACTIVATION_MS */,
+    },
+  },
+  rippleDisabled: true
+};
+
+export const MAT_MDC_CHECKBOX_CONTROL_VALUE_ACCESSOR: any = {
```

Comment:
You should be able to get of the any by typing it to a `Provider`.

Reply:
I'll keep it `any` for now to match what's in `src/lib/checkbox`. Also now that you point it out, it looks like I forgot to remove the `MDC_` from the name, fixed.